src/app/frontend/attach/controller.js

```diff
@@ -0,0 +1,163 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {stateName as attach, StateParams} from './state';
+
+/**
+ * Controller for the attach view.
+ *
+ * @final
+ */
+export class AttachController {
+  /**
+   * @param {!backendApi.PodContainerList} podContainers
+   * @param {!angular.$sce} $sce
+   * @param {!angular.$document} $document
+   * @param {!./state.StateParams} $stateParams
+   * @param {!angular.$resource} $resource
+   * @param {!ui.router.$state} $state
+   * @ngInject
+   */
+  constructor(podContainers, $sce, $document, $resource, $stateParams, $state) {
+    /** @private {!angular.$sce} */
+    this.sce_ = $sce;
+
+    /** @private {!HTMLDocument} */
+    this.document_ = $document[0];
+
+    /** @private {!angular.$resource} */
+    this.resource_ = $resource;
+
+    /** @private {!./state.StateParams} $stateParams */
+    this.stateParams_ = $stateParams;
+
+    /** @export */
+    this.i18n = i18n;
+
+    /** @private {!ui.router.$state} */
+    this.state_ = $state;
+
+    /** @export {!Array<string>} */
+    this.containers = podContainers.containers ? podContainers.containers : [];
+
+    /** @export {string} */
+    this.container = this.stateParams_.container ? this.stateParams_.container : '';
+
+    /** @export {string} */
+    this.podName = this.stateParams_.objectName ? this.stateParams_.objectName : '';
+
+    /** private {hterm.Terminal} */
+    this.term = null;
+
+    /** @private {SockJS} */
+    this.conn = null;
+
+    this.prepareTerminal();
+  }
+
+  prepareTerminal() {
+    let self = this;
+
+    let namespace = this.stateParams_.objectNamespace;
+    let podName = this.stateParams_.objectName;
+    let containerName = this.stateParams_.container || '';
+
+
+    // https://chromium.googlesource.com/apps/libapps/+/HEAD/hterm/doc/embed.md
+    hterm.defaultStorage = new lib.Storage.Memory();
+
+    // https://github.com/sockjs/sockjs-client
+
+    this.term = new hterm.Terminal();
+    this.term.onTerminalReady = function() {
```

Comment:
I'd extract anonymous functions to class functions. It will be easier to document and understand the code.

By the way instead of `function() {}` you can use ES6 fat arrow syntax `() => {}`.